modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -85,6 +85,10 @@ template<typename _Tp> inline
 _InputArray::_InputArray(const std::vector<std::vector<_Tp> >& vec)
 { init(FIXED_TYPE + STD_VECTOR_VECTOR + DataType<_Tp>::type + ACCESS_READ, &vec); }
 
+inline
+_InputArray::_InputArray(const std::vector<std::vector<bool> >&)
+{ CV_Error(Error::StsUnsupportedFormat, "std::vector<std::vector<bool> > is not supported!\n"); }
```

Comment:
[CV_StaticAssert](https://github.com/opencv/opencv/blob/3.2.0/modules/core/include/opencv2/core/matx.hpp#L503) would be better here because it is not a "runtime" problem.